Gemfile

```diff
@@ -4,27 +4,38 @@ git_source(:github) do |repo_name|
   repo_name = "#{repo_name}/#{repo_name}" unless repo_name.include?('/')
   "https://github.com/#{repo_name}.git"
 end
+
+gem 'bootstrap', '~> 4.0.0'
+gem 'cancancan', '~> 2.0'
 gem 'coffee-rails', '~> 4.2'
+gem 'devise', '~> 4.4.3'
+gem 'haml', '~> 2.0.0'
 gem 'jbuilder', '~> 2.5'
-gem 'mysql2', '>= 0.3.18', '< 0.5'
-gem 'puma', '~> 3.7'
 gem 'rails', '~> 5.1.5'
+gem 'rolify', '~> 5.2.0'
 gem 'sass-rails', '~> 5.0'
 gem 'turbolinks', '~> 5'
 gem 'uglifier', '>= 1.3.0'
 
 group :development, :test do
   gem 'byebug', platforms: %i[mri mingw x64_mingw]
   gem 'capybara', '~> 2.13'
-  gem 'rspec-rails'
-  gem 'selenium-webdriver'
+  gem 'factory_bot', '~> 4.0'
+  gem 'mysql2', '>= 0.3.18', '< 0.5'
+  gem 'puma', '~> 3.7'
+  gem 'rspec-rails', '~> 3.7.0'
+  gem 'selenium-webdriver', '~> 3.11.0'
 end
 
 group :development do
   gem 'listen', '>= 3.0.5', '< 3.2'
-  gem 'spring'
+  gem 'spring', '~> 2.0.0'
   gem 'spring-watcher-listen', '~> 2.0.0'
   gem 'web-console', '>= 3.3.0'
 end
 
+group :production do
+  gem 'unicorn', '~> 5.3.0'
```

Comment:
I do not see any DMS you are going to use in production. Is there any reason you did not add it?